lib/saml_idp/encryptor.rb

```diff
@@ -34,8 +34,25 @@ def openssl_cert
     end
     private :openssl_cert
 
+    # Encryption algorithms enumerated in:
+    # https://github.com/digidentity/xmlenc/blob/937ca2f/lib/xmlenc/encrypted_data.rb#L3-L10
     def block_encryption_ns
-      "http://www.w3.org/2001/04/xmlenc##{block_encryption}"
+      case block_encryption
+        when 'tripledes-cbc'
+          'http://www.w3.org/2001/04/xmlenc#tripledes-cbc'
+        when 'aes128-cbc'
+        'http://www.w3.org/2001/04/xmlenc#aes128-cbc'
+        when 'aes256-cbc'
+          'http://www.w3.org/2001/04/xmlenc#aes256-cbc'
+        when 'aes128-gcm'
+          'http://www.w3.org/2009/xmlenc11#aes128-gcm'
+        when 'aes192-gcm'
+          'http://www.w3.org/2009/xmlenc11#aes192-gcm'
+        when 'aes256-gcm'
+          'http://www.w3.org/2009/xmlenc11#aes256-gcm'
+        else
+          "http://www.w3.org/2009/xmlenc11##{block_encryption}"
+      end
```

Comment:
this indentation is not what I expect? did Rubocop do this?

```suggestion
      case block_encryption
      when 'tripledes-cbc'
        'http://www.w3.org/2001/04/xmlenc#tripledes-cbc'
      when 'aes128-cbc'
      'http://www.w3.org/2001/04/xmlenc#aes128-cbc'
      when 'aes256-cbc'
        'http://www.w3.org/2001/04/xmlenc#aes256-cbc'
      when 'aes128-gcm'
        'http://www.w3.org/2009/xmlenc11#aes128-gcm'
      when 'aes192-gcm'
        'http://www.w3.org/2009/xmlenc11#aes192-gcm'
      when 'aes256-gcm'
        'http://www.w3.org/2009/xmlenc11#aes256-gcm'
      else
        "http://www.w3.org/2009/xmlenc11##{block_encryption}"
      end
```

Reply:
I switched to put all of the algorithms and values in a hash constant so it's easier to shorten this up and make it easier to reference all of them in tests.